### PR TITLE
Bug 1357875 - Use toLocalIterator instead of writing directly to `file://` via spark

### PR DESCRIPTION
**Pull request by acmiyaguchi**
There are certain permission issues when deploying topline_dashboard due to permission errors with the spark executor. One alternative is to write the csv files to hdfs and copy over to local disk. This is not necessarily an elegant solution to the problem, so the tried solution is to use the collect (as long as this dataset stays within the size of the driver).

This patch works in practice, and was used to generate reports under `net-mozaws-prod-us-west-2-pipeline-analysis/amiyaguchi/topline-dashboard-data/`

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/python_mozetl/27)

### CODE REVIEW

File: mozetl/topline/topline_dashboard.py [hunk 1]
```diff
@@ -122,16 +122,17 @@ def write_dashboard_data(df, bucket, prefix, mode):
...
+
+    with open(filepath, 'wb') as f:
+        f.write(','.join(df.columns) + '\n')
+        rows = df.collect()
+        for row in rows:
+            try:
+                f.write(','.join([unicode(c) for c in row]))
+                f.write('\n')
+            except UnicodeEncodeError:
+                logger.exception("exception while processing csv")
```
Comment (mreid-moz) [inline mozetl/topline/topline_dashboard.py]: I don't know if it's a good idea to potentially drop rows with just a logged message. Does this happen in practice? If so, under what circumstances?

Comment (acmiyaguchi) [inline mozetl/topline/topline_dashboard.py]: This doesn't happen in practice because the topline_summary is generally well formed. I don't don't think there are malformed strings in the original data.

Comment (mreid-moz) [inline mozetl/topline/topline_dashboard.py]: Ok, can you confirm that writing bare 'strings' (rather than u'unicode' strings) works ok in combination with unicode field values?

It might be worthwhile to factor out some "write this dataframe to csv" functionality and add some tests. handling all the unicode/utf8 corner cases in python2 always seems to be fraught with peril.

**Comment by acmiyaguchi** [comment]
The source of the failing tests is from a bad interaction of pyspark rdd methods with moto. I've patched the offending methods.

I've also put the csv writer method into a utility module. I still need to write a few tests that verify the behavior with unicode/utf-8.

File: mozetl/utils.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import csv
+import pdb
+
+
+def write_csv(dataframe, path):
+    """ Write a dataframe to local disk. """
+
+    # pdb.set_trace()
```
Comment (mreid-moz) [inline mozetl/utils.py]: Can this be removed (along with the import above)?

**Comment by acmiyaguchi** [comment]
@mreid-moz RFAL
